opentargets_pharmgkb/ontology_apis.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import os
+from functools import lru_cache
+
+import requests
+from cmat.trait_mapping.main import process_trait
+from cmat.trait_mapping.trait import Trait
+from requests import RequestException
+from retry import retry
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+HOST = 'https://www.ebi.ac.uk'
+OLS_API_ROOT = f'{HOST}/ols/api'
+
+
+@lru_cache
+@retry(exceptions=(ConnectionError, RequestException), tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+def get_chebi_iri(drug_name):
+    chebi_search_url = os.path.join(OLS_API_ROOT, f'search?ontology=chebi&q={drug_name}')
+    response = requests.get(chebi_search_url)
+    response.raise_for_status()
+    data = response.json()
+    if 'response' in data:
+        results = data['response']['docs']
+        candidates = set()
+        for result in results:
+            # Check that we've found the drug exactly (strict case-insensitive string match)
+            if result['label'].lower() == drug_name.lower():
```

Comment:
@apriltuesday, since you have dealt a bit more with PharmGKB, and just to confirm, their drug labels are always in concordance with CHEBI labels, right? As in, in the same language and everything. 

Asking because I saw **so** many synonyms for some examples I checked, that I got worried. These examples, nevertheless, were matching perfectly with CHEBI.

Reply:
They're not always the same name, an example from the tests is `fluorouracil` which is labelled `5-fluorouracil` in CHEBI (http://purl.obolibrary.org/obo/CHEBI_46345) and thus doesn't come up as a match in OLS.

This is why we also [check PGKB primary data](https://github.com/EBIvariation/opentargets-pharmgkb/blob/main/opentargets_pharmgkb/evidence_generation.py#L91), so we can pick up some of these cases.  (The reason we don't always do this is, of course, PGKB does not always include CHEBI IDs... that would make things _too_ easy for us...)

Reply:
No no, I was aware of the fluorouracil test, I wondered that, **if** a compound exists in CHEBI, its **main** label is the same **main** label that PharmaGKB uses. There can be multiple synonyms, but since we check the label, I was just hoping they use the same main one.

I assume ``fluorouracil`` and  ``5-fluorouracil`` are different compounds, so it shouldn't match. But the other case, where different synonyms are used is what I was afraid of. But in any case you answered the question, thanks!